### PR TITLE
stylo: fix border-radius longhand specified values serialization

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fixes part of [bug 1397619](https://bugzilla.mozilla.org/show_bug.cgi?id=1397619).
- [X] There are tests for these changes attached to bugzilla

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18458)

### CODE REVIEW

File: components/style/values/specified/border.rs [hunk 1]
```diff
@@ -132,27 +132,33 @@ impl Parse for BorderImageSlice {
...
 
 impl Parse for BorderCornerRadius {
     fn parse<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
         let first = LengthOrPercentage::parse_non_negative(context, input)?;
+        let mut heights_specified = true;
         let second = input
             .try(|i| LengthOrPercentage::parse_non_negative(context, i))
-            .unwrap_or_else(|_| first.clone());
-        Ok(Self::new(first, second))
+            .unwrap_or_else(|_| {
```
Comment (emilio) [inline components/style/values/specified/border.rs]: This looks wrong to me. Should instead avoid using `Size2D`, and using:

```rust
struct SpecifiedBorderRadius {
    width: specified::LengthOrPercentage,
    height: Option<specified::LengthOrPercentage>,
}
```

and

```rust
struct ComputedBorderRadius {
    width: computed::LengthOrPercentage,
    height: computed::LengthOrPercentage,
}
```

@nox, do you know something off-hand to be able to derive those?

**Review by emilio — CHANGES_REQUESTED** [review]

File: components/style/values/specified/border.rs (hunk 1 above)
Comment (emilio) [reply components/style/values/specified/border.rs]: (I mean, `derive(ToCss)` should work automatically, but not `derive(ToComputedValue)`, I think.

Comment (nox) [reply components/style/values/specified/border.rs]: I have an idea where I want to do `OmittedDefault<T, D>(Option<T>, PhantomData<Default>)` where `D` is `Default` and `Into<T>`.

**Pull request closed without merging**